lib/script/script.js

```diff
@@ -713,14 +727,44 @@ Script.fromAddress = function(address) {
 
 /**
  * @param {Network=} network
- * @return {Address} the associated address for this script
+ * @return {Address|boolean} the associated address information object
+ * for this script if any, or false
+ */
+Script.prototype.getAddressInfo = function() {
+  var Address = require('../address');
+  var info = {};
+  if (this.isScriptHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToScriptHash;
+  } else if (this.isPublicKeyHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToPublicKeyHash;
+  } else if (this.isPublicKeyHashIn()) {
+    // hash the publickey found in the scriptSig
+    info.hashBuffer = Hash.sha256ripemd160(this.chunks[1].buf);
```

Comment:
Thanks for the merge!

On Thu, Mar 19, 2015, 09:12 Esteban Ordano <notifications@github.com> wrote:

> In lib/script/script.js
> <https://github.com/bitpay/bitcore/pull/1151#discussion_r26748074>:
>
> > - * @return {Address} the associated address for this script
> > + * @return {Address|boolean} the associated address information object
> > + * for this script if any, or false
> > + */
> > +Script.prototype.getAddressInfo = function() {
> > +  var Address = require('../address');
> > +  var info = {};
> > +  if (this.isScriptHashOut()) {
> > +    info.hashBuffer = this.getData();
> > +    info.type = Address.PayToScriptHash;
> > +  } else if (this.isPublicKeyHashOut()) {
> > +    info.hashBuffer = this.getData();
> > +    info.type = Address.PayToPublicKeyHash;
> > +  } else if (this.isPublicKeyHashIn()) {
> > +    // hash the publickey found in the scriptSig
> > +    info.hashBuffer = Hash.sha256ripemd160(this.chunks[1].buf);
>
> Sounds good!
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore/pull/1151/files#r26748074>.
>